src/main/java/org/opensearch/security/securityconf/impl/SecurityDynamicConfiguration.java

```diff
@@ -322,7 +314,19 @@ public boolean containsAny(SecurityDynamicConfiguration other) {
 
     public boolean isHidden(String resourceName) {
         final Object o = centries.get(resourceName);
-        return o != null && o instanceof Hideable && ((Hideable) o).isHidden();
+        return o instanceof Hideable && ((Hideable) o).isHidden();
+    }
+
+    @JsonIgnore
+    public boolean isStatic(final String resourceName) {
+        final Object o = centries.get(resourceName);
+        return o instanceof StaticDefinable && ((StaticDefinable) o).isStatic();
+    }
+
+    @JsonIgnore
+    public boolean isReserved(final String resourceName) {
+        final Object o = centries.get(resourceName);
+        return o instanceof Hideable && ((Hideable) o).isReserved();
```

Comment:
Looking into this method of `AbstractApiAction::isReserved` that says static is also always reserved, should we baked it into the `isReserved` check?

Reply:
It is the same in the `AbstractApiAction` class. The PR fixes broken responsibility since teh configuration has a predefined reserved and static fields. E.g. `isHidden` was moved while `isStatic` and `isReseved` not

Reply:
Ok, let me try to ask the question this way:
 - why `AbstractApiAction::isReserved` checks `configuration.isStatic(resourceName) || configuration.isReserved(resourceName)`
 - but `SecurityDynamicConfiguration::isReserved` is only checking `isReserved` ?

May be `AbstractApiAction::isReserved` should be renamed or `SecurityDynamicConfiguration::isReserved` should include `isStatic` check as well?

Reply:
Because it is forbidden to change any `static`, `reserved` and `hidden` resource for non REST admin or Super admin user. TBH I'm planning to change the name after the review of the https://github.com/opensearch-project/security/pull/2900 has been finished.